uitools/toolkitcpp/import/Esri/ArcGISRuntime/Toolkit/MediaPopupElementView.qml

```diff
@@ -43,74 +44,221 @@ ColumnLayout {
+                if (popupMediaType === QmlEnums.PopupMediaTypeImage) {
+                    loader.sourceComponent = imageComp;
+                } else if (popupMediaType === QmlEnums.PopupMediaTypeColumnChart) {
+                    loader.sourceComponent = columnChartComp;
+                } else if (popupMediaType === QmlEnums.PopupMediaTypeBarChart) {
+                    loader.sourceComponent = barChartComp;
+                } else if (popupMediaType === QmlEnums.PopupMediaTypePieChart) {
+                    loader.sourceComponent = pieChartComp;
+                } else if (popupMediaType === QmlEnums.PopupMediaTypeLineChart) {
+                    loader.sourceComponent = lineChartComp;
+                }
+            }
+
+            Component {
+                id: imageComp
+                Image {
+                    source: listModelData.sourceUrl
+                    fillMode: Image.PreserveAspectCrop
+                    asynchronous: true
+                    cache: true
+                    height: delegatePopupMedia.height
+                    width: delegatePopupMedia.width
+
+                    Layout.leftMargin: mediaPopupElementView.mediaMargin
+
+                    MouseArea {
+                        anchors.fill: parent
+                        onClicked: {
+                            // saving this as a placeholder. Should fire an event the user could take otherwise
+                            // we define our own default behavior to open the image in full screen mode.
+                            // Qt.openUrlExternally(model.linkUrl);
+                        }
+                    }
+                }
+            }
+
+            Component {
+                id: columnChartComp
+                GraphsView {
+                    height: delegatePopupMedia.height
+                    width: delegatePopupMedia.width
+                    marginTop: 10
+                    marginBottom: 10
+                    marginLeft: 10
+                    marginRight: 10
+
+                    // Setting a default theme if no color information is provided on the BarSeries
+                    theme: GraphsTheme {
+                        backgroundColor: "white"
+                        plotAreaBackgroundColor: "white"
+                        theme: GraphsTheme.Theme.MixSeries
+                    }
+
+                    axisY: ValueAxis {
+                        max: listModelData.maxValue
+                        min: listModelData.minValue
+                        tickInterval: Math.abs(max) > Math.abs(min)? max / 3 : Math.abs(min) / 3
+                    }
+
+                    BarSeries {
+                        // BarSeries will take owership of the QList<QBarSet*> when we call append. This also applies to PieSeries and QPieSlice for PieChartPopupMediaItem.
+                        // s.a https://github.com/qt/qtgraphs/blob/ef90c963d10a1af30ed523483e9a589235eccef9/src/graphs2d/barchart/qbarseries.cpp#L670-L692
```

Comment:
Is this documented, or only discovered by digging through the sources? We should work with Qt to get this documented.

Reply:
It looks like it is but not on the QML side of the documentation just the C++ side. Ill update the link to the C++ documentation which does indicate that function is available via QML